File: wclib/src/main/java/com/yyc/wclib/mvp/MvpBaseActivity.java
/*
package com.yyc.wclib.mvp;

import android.os.Bundle;

import com.yyc.wclib.base.BaseActivity;
import com.yyc.wclib.utils.CreatUtilOne;

import org.jetbrains.annotations.Nullable;

public abstract class MvpBaseActivity<P extends BasePresenter> extends BaseActivity{
    public P mPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        mPresenter = CreatUtilOne.getT(this,0);
        mPresenter.attachView(this);
        super.onCreate(savedInstanceState);

    }

*/
/*
    @Override
    public void initView() {

    }
*//*


    @Override
    public void initData() {

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mPresenter.dettachView();
    }
}
*/

File: wclib/src/main/java/com/yyc/wclib/mvp/MvpBaseActivity.kt
package com.yyc.wclib.mvp

import android.os.Bundle
import android.util.Log
import com.yyc.baselib.utils.CreatUtil
import com.yyc.wclib.base.BaseActivity
abstract class MvpBaseActivity<V: BaseView,P : BasePresenter2<V,*> >: BaseActivity(){
    var mPresenter: P? =null//可空类型

    override fun onCreate(savedInstanceState: Bundle?) {
//        mPresenter = CreatUtilOne.getT(this,0)
        mPresenter = CreatUtil.getT(this,0)
        super.onCreate(savedInstanceState)
        Log.e("aa", "mPresenter: $mPresenter")
//        mPresenter?.attachView(this)
        mPresenter!!.bindView(this as V)
    }

    public abstract fun initPresenter(): P

    override fun initView() {
    }

    override fun initData() {
    }

    override fun onDestroy() {
        super.onDestroy()
//        mPresenter?.dettachView()
    }
}
